lib/chromecastDevice.js

```diff
@@ -907,13 +909,37 @@ module.exports = function (adapter) {
                 return;
             }
 
-            this._mediaPlayer.playUrlPromise(url2play /*org_url2play, streamType*/)
-                .then(() => {
-                    adapter.setState(`${this.mac}.player.url2play`, {val: org_url2play, ack: true});
+            // Is this a Youtube URL?
+            const youtubeId = Youtube.getYoutubeId(url2play)
+            if (youtubeId) {
+                adapter.log.debug('Trying to play YouTube (' + youtubeId + ') on ' + this._connection.name);
+                const client = this._mediaPlayer._client._client;
+
+                // TODO: do something funky with sessions & re-use?
```

Comment:
As  I elude to... think something fishy is going on with repeat attempts to play videos and more work required here.